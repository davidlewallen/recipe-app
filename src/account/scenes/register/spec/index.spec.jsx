import React from 'react';
import { shallow } from 'enzyme';

import Register from '../components';

describe('Register component snapshot test', () => {
  const props = {
    password: 'testPassword',
    passwordConf: 'testPassword',
    email: 'test@email.com',
    register: jest.fn(),
    error: { value: false, message: '' },
    handleInputChange: jest.fn(),
  };

  it('should match snapshot', () => {
    const component = shallow(<Register {...props} />);

    expect(component).toMatchSnapshot();
  });

  it('should display error messages', () => {
    const component = shallow(
      <Register {...props} error={{ value: true, message: 'testError' }} />
    );

    expect(component).toMatchSnapshot();
  });
});
