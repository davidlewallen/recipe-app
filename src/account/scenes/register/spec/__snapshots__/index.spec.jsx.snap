// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register component snapshot test should display error messages 1`] = `
<Grid
  bsClass="container"
  className="register-container"
  componentClass="div"
  fluid={false}
>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      sm={6}
      smOffset={3}
      xs={12}
    >
      <form
        className="register"
      >
        <FormGroup
          bsClass="form-group"
        >
          <p>
            testError
          </p>
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            Username
          </ControlLabel>
          <FormControl
            bsClass="form-control"
            className="username-input"
            componentClass="input"
            onChange={[Function]}
            placeholder="E.g. JohnDoe123"
            type="text"
            value="testUsername"
          />
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            Password
          </ControlLabel>
          <FormControl
            bsClass="form-control"
            className="password-input"
            componentClass="input"
            onChange={[Function]}
            placeholder="E.g. 124!Jigzx"
            type="password"
            value="testPassword"
          />
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            Email Address
          </ControlLabel>
          <FormControl
            bsClass="form-control"
            className="email-input"
            componentClass="input"
            onChange={[Function]}
            placeholder="E.g. john@doe.com"
            value="test@email.com"
          />
          <Button
            active={false}
            block={true}
            bsClass="btn"
            bsStyle="primary"
            className="register-button"
            disabled={false}
            onClick={[Function]}
            type="submit"
          >
            Register
          </Button>
          <div
            className="already-registered-text"
          >
            Already have an account? 
            <Link
              replace={false}
              to="/account/login"
            >
              Sign in here.
            </Link>
          </div>
        </FormGroup>
      </form>
    </Col>
  </Row>
</Grid>
`;

exports[`Register component snapshot test should match snapshot 1`] = `
<Grid
  bsClass="container"
  className="register-container"
  componentClass="div"
  fluid={false}
>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      sm={6}
      smOffset={3}
      xs={12}
    >
      <form
        className="register"
      >
        <FormGroup
          bsClass="form-group"
        >
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            Username
          </ControlLabel>
          <FormControl
            bsClass="form-control"
            className="username-input"
            componentClass="input"
            onChange={[Function]}
            placeholder="E.g. JohnDoe123"
            type="text"
            value="testUsername"
          />
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            Password
          </ControlLabel>
          <FormControl
            bsClass="form-control"
            className="password-input"
            componentClass="input"
            onChange={[Function]}
            placeholder="E.g. 124!Jigzx"
            type="password"
            value="testPassword"
          />
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            Email Address
          </ControlLabel>
          <FormControl
            bsClass="form-control"
            className="email-input"
            componentClass="input"
            onChange={[Function]}
            placeholder="E.g. john@doe.com"
            value="test@email.com"
          />
          <Button
            active={false}
            block={true}
            bsClass="btn"
            bsStyle="primary"
            className="register-button"
            disabled={false}
            onClick={[Function]}
            type="submit"
          >
            Register
          </Button>
          <div
            className="already-registered-text"
          >
            Already have an account? 
            <Link
              replace={false}
              to="/account/login"
            >
              Sign in here.
            </Link>
          </div>
        </FormGroup>
      </form>
    </Col>
  </Row>
</Grid>
`;
